tests/__snapshots__/settings.spec.ts.snap

```diff
@@ -10,34 +10,37 @@ LOG FILTERING SYNTAX MANUAL  ⟁
 
 Grammar
 
-    Precise: 
-
-    /^(!)?((?:[A-z_]+:)+)?(?:((:)?\\\\*)?|([A-z_]+|))?(?<=\\\\*|[A-z_]+)(?:(?:@(1|2|3|4|5|6|trace|debug|info|warn|error|fatal)([-+]?))|(@\\\\*))?$/
```

Comment:
I spent at least an hour trying to integrate root syntax into here but gave up. Anyways one of the downsides of regex AFAIK is that there is no way to give a nice error message about why something didn't match to the user. So I guess in the long run we don't want regex anyways. But it sure is nice to have the entire parser self-contained in a nice succinct program like that <3 

While the regex might feel hard to read, my custom parsing code isn't much better!